urllib3/contrib/pyopenssl.py

```diff
@@ -127,6 +129,17 @@ def extract_from_urllib3():
     util.ssl_.IS_PYOPENSSL = False
 
 
+def _validate_dependencies_met():
+    """
+    Verifies that PyOpenSSL's package-level dependencies have been met.
+    Throws `ImportError` if they are not met.
+    """
+    # Method added in `cryptography==1.1`; not available in older versions
+    from cryptography.x509.extensions import Extensions
+    if not hasattr(Extensions, "get_extension_for_class"):
+        raise ImportError("'cryptography' module missing required function (try upgrading the module?): 'cryptography.x509.extensions.Extensions.get_extension_for_class'")
```

Comment:
As another side note, I think this only shows that cryptography meets version 1.1. Is there any way we can modify the validation so we meet the 1.3.4 requirement in setup.py? That way we don't have a weird limbo state in 1.2-1.3.3 were this passes but we may still be missing dependencies.